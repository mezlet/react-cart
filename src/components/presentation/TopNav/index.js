import React from 'react'

const TopNav = () => {
    return (
        <div className='p-5 f-12'>
            <h1 className='px-5 text-secondary'>All Products</h1>
            <div className='d-flex justify-content-between'>
                <span className='px-5 text-secondary'>A 360 look at Lumin</span>
                <select name="" id="" className='pr-5'>
                    <option value="" className='text-secondary'>Filter By</option>
                </select>
            </div>
        </div>
    )
}

export default TopNav
