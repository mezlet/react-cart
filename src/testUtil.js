import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
    return checkPropTypes(component.propTypes, expectedProps, 'props', component.name)
}