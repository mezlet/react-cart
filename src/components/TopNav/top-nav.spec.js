import React from 'react';
import { shallow } from 'enzyme';
import TopNav from './index';
import { checkProps} from '../../testUtil';

const props = {}

const setUp = (props = {}) => {
    const component = shallow(<TopNav {...props} />);
    return component;
}

describe('TopNav', () => {
    let component;
    beforeEach(() => {
        component = setUp(props);
    })
    it('should render without errors', () => {
        expect(component).toMatchSnapshot(); 
    });
    it('should render top nav div', () => {
        const wrapper = component.find(`[data-test='top-nav-div']`);
        expect(wrapper.length).toBe(1); 
    });
})