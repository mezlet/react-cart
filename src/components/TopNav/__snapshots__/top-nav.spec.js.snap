// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav should render without errors 1`] = `ShallowWrapper {}`;
